### PR TITLE
Split out v2 packages

### PR DESCRIPTION
**Pull request by infiloop2**
- Move packages in /pkg to /v2 to have a clear split between v2 and v3
- only utils package is shared

- config was shared between packages, copy it into both v2 and v3. Ensure files in v2/folder refer to v2 config, and in v3 folder refer to v3 config

### CODE REVIEW

File: cmd/simv3/main.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ import (
 	"github.com/smartcontractkit/ocr2keepers/cmd/simv3/run"
 	"github.com/smartcontractkit/ocr2keepers/cmd/simv3/simulators"
 	"github.com/smartcontractkit/ocr2keepers/cmd/simv3/telemetry"
-	"github.com/smartcontractkit/ocr2keepers/pkg/encoding"
+	"github.com/smartcontractkit/ocr2keepers/pkg/v2/encoding"
```
Comment (infiloop2) [inline cmd/simv3/main.go:23]: not sure why v2 encoding was referenced in simv3, ignoring it for now

File: cmd/simv3/simulators/registry_test.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
 
-	"github.com/smartcontractkit/ocr2keepers/pkg/encoding"
+	"github.com/smartcontractkit/ocr2keepers/pkg/v2/encoding"
```
Comment (infiloop2) [inline cmd/simv3/simulators/registry_test.go:12]: same. not sure why v2 encoding was referenced in simv3, ignoring it for now

File: pkg/v3/coordinator/conditional_report_coordinator.go [hunk 3]
```diff
@@ -42,7 +42,7 @@ var (
 	cadence               = time.Second
 )
 
-//go:generate mockery --name Encoder --srcpkg "github.com/smartcontractkit/ocr2keepers/pkg/coordinator" --case underscore --filename encoder.generated.go
+//go:generate mockery --name Encoder --srcpkg "github.com/smartcontractkit/ocr2keepers/pkg/v3/coordinator" --case underscore --filename encoder.generated.go
```
Comment (infiloop2) [inline pkg/v3/coordinator/conditional_report_coordinator.go:45]: this was incorrect before

File: cmd/simv3/main.go (hunk 1 above)
Comment (ferglor) [reply cmd/simv3/main.go:23]: The few v2 components in the v3 sim that I saw as part of the type refactoring, weren't actually being used, so I guess it was maybe just a copy-paste artifact?

**Review by ferglor — DISMISSED** [review]

File: cmd/simv3/main.go (hunk 1 above)
Comment (infiloop2) [reply cmd/simv3/main.go:23]: i tried removing them but ran into errors so let them be for now

**Review by amirylm — APPROVED** [review]

File: cmd/simv3/main.go (hunk 1 above)
Comment (infiloop2) [reply cmd/simv3/main.go]: actually was able to remove them now i think :D

**Review by ferglor — APPROVED** [review]

**Pull request merged**